site/joseki.js

```diff
@@ -545,31 +589,76 @@ const EMPTY_SCORE = {
                 currentEditGroupIndex = groupIndex;
                 let moves = currentEditJoseki.moves;
                 currentEditJoseki.moves = [];
-                for (const move of moves){
-                    if(move == PASS){
-                        editPass();
+                for (const moveStr of moves){
+                    let move = parseMove(moveStr);
+                    if(move.type == PASS){
+                        editPass(move.isAuto);
                     } else {
-                        let [x,y] = parseMove(move);
-                        handleEditAdd(x,y);
+                        editAdd(move.x, move.y, move.isAuto);
                     }
                 }
                 document.getElementById('comment').value = currentEditJoseki.comment;
                 josekiGroupSelectEl.value = groupIndex;
             }
         }
+
+        setAllowAutoStone();
+    }
+
+    function handleEditStoneTypeChange() {
+        let stoneType = document.getElementById('editStoneType')?.value;
+        let isAuto = stoneType === 'automatic';
+        editStoneTypeChange(isAuto);
+    }
+
+    function editStoneTypeChange(isAuto) {
+        let passButton = document.getElementById('passButton');
+        passButton.innerText = isAuto ? 'Automatic Pass' : 'Pass / Tenuki';
+    }
+
+    function setAllowAutoStone() {
+        if (josekiData?.userSettings?.allowUnrestrictedAutoStones)
+            return;
+
+        // Setup/Auto stones can only start a joseki.  They cannot be after a non-automatic move.
```

Comment:
This isn't true - I was able to switch to "setup" stones after adding manual moves, creating that surprising situation where the cpu takes over during a joseki

Reply:
I am surprised by this?  I can't seem to recreate on Chrome or FF.  Are we sure your local code is fully pulled?  I force pushed this branch before messaging you

Reply:
Oh well that is exactly why then, I didn't notice the setting.